source/dhtslib/coordinates.d

```diff
@@ -39,6 +39,125 @@ obc  -1,0   -1,-1    0,+1    -
 */
 
 module dhtslib.coordinates;
+import std.conv : to;
+
+/// Represents 0-based vs 1-based coordinate types
+enum Basis
+{
+    zero = 0,
+    one
+}
+
+/// Represents whether a coordinate set's end 
+/// coordinate is open or closed
+enum End
+{
+    open = 0,
+    closed
+}
+
+/// Coordinate type for a single position with in a Coordinate set,
+/// where the type itself encodes the coordinate system details
+/// (zero or one-based)
+struct Coordinate(Basis bs)
+{
+    long pos;
+    alias pos this;
```

Comment:
`alias this` may be deprecated IIRC, please check docs and forum

Reply:
Doesn't appear so from what I saw.